src/task.ts

```diff
@@ -92,6 +92,12 @@ export default class Task {
     return this.instanceAt(name, address);
   }
 
+  async deployedInstanceDifferentArtifact(name: string, artifactName: string): Promise<Contract> {
+    const address = this.output()[name];
+    if (!address) throw Error(`Could not find deployed address for ${name}`);
+    return this.instanceAt(artifactName, address);
+  }
+
```

Comment:
```suggestion
```
See how to implement this in a standard way below. (I don't think it would work as is.)